Source/WebCore/html/HTMLSourceElement.cpp

```diff
@@ -219,4 +219,22 @@ void HTMLSourceElement::addCandidateSubresourceURLs(ListHashSet<URL>& urls) cons
     getURLsFromSrcsetAttribute(*this, attributeWithoutSynchronization(srcsetAttr), urls);
 }
 
+Ref<Element> HTMLSourceElement::cloneElementWithoutAttributesAndChildren(Document& targetDocument)
+{
+    auto clone = create(targetDocument);
+#if ENABLE(ATTACHMENT_ELEMENT)
+    cloneAttachmentAssociatedElementWithoutAttributesAndChildren(clone, targetDocument);
+#endif
+    return clone;
+}
+
+void HTMLSourceElement::copyNonAttributePropertiesFromElement(const Element& source)
+{
+#if ENABLE(ATTACHMENT_ELEMENT)
+    auto& sourceElement = static_cast<const HTMLSourceElement&>(source);
+    copyAttachmentAssociatedPropertiesFromElement(sourceElement);
```

Comment:
Could this use `downcast<>` or `dynamicDowncast<>` instead?

Reply:
Used `checkedDowncast`.